.gitignore

```diff
@@ -2,3 +2,10 @@ build/*
 CMakeUserPresets.json
 cppcheck-*
 *.cppcheck
+
+# Tim Littlefair, October 2024
+# When I created a kdevelop project, kdevelop created a directory 
+# .kdev4 which contains a single file .kdev4/offa-plug.kdev4. 
+# As this file contains the full path my local build directory
+# I'm guessing that it should not be committed to git.
+.kdev4
```

Comment:
Feel free to submit this line directly, extending `.gitignore` might help others too.

Reply:
I'm not sure what the invitation is here - do you want me to create a PR with the file ./offa-plug.kdev4 in it, and whatever else is required so that the kdevelop project works out of the box on a fresh clone?

I can't see the point of adding a line for the .kdev4 subdirectory to .gitignore  unless it is alongside the project file.     I added this because the files the in .kdev4 subdirectory contained information the full path to the sandbox directory on my computer, so I'm guessing that these files are not particularly helpful to others (and might be harmful).

Experimentation shows that adding the offa-plug.kdev4 file only doesn't seem to give a clean startup so I'll look into what is required for a path-neutral project and I'm happy to do a PR containing this if it is what is preferred.

Reply:
> I'm not sure what the invitation is here - do you want me to create a PR with the file ./offa-plug.kdev4 in it, and whatever else is required so that the kdevelop project works out of the box on a fresh clone?

No, just add project files to the .gitignore so others that might use the IDE don't commit their local files accidentally.